docs/source/en/tasks_explained.mdx

```diff
@@ -0,0 +1,230 @@
+
+## Natural language processing
+
+The Transformer was originally designed for machine translation, and since then, it has practically become the default architecture for solving all NLP tasks. Some tasks lend themselves to the Transformer's encoder structure, while others are better suited for the decoder. Still, other tasks make use of both the Transformer's encoder-decoder structure.
+
+### Text classification
+
+[BERT](model_doc/bert) is an encoder-only model and is the first model to effectively implement deep bidirectionality, which allows BERT to learn richer representations of the text by attending to words before and after.
+
+1. BERT uses [WordPiece](tokenizer_summary#wordpiece) tokenization to generate a token embedding of the text. To tell the difference between a single sentence and a pair of sentences, a special `[SEP]` token is added to differentiate them. A special `[CLS]` token is added to the beginning of every sequence of text. The final output with the `[CLS]` token is used as the input to the classification head for classification tasks. BERT also adds a segment embedding to denote whether a token belongs to the first or second sentence in a pair of sentences.
+
+2. BERT is pretrained with two objectives: masked language modeling and next sentence prediction. In masked language modeling, some percentage of the input tokens are randomly masked, and the model needs to predict these. This solves the issue of deep bidirectionality, where the model could cheat and see all the words and "predict" the next word. The final hidden states of the predicted mask tokens are passed to a feedforward network with a softmax over the vocabulary to predict the masked word.
+
+    The second pretraining object is next sentence prediction. The model must predict whether sentence B is the next sentence after sentence A. Half of the time sentence B is the next sentence and the other half of the time, sentence B is a random sentence. The prediction, whether it is the next sentence or not, is passed to a feedforward network with a softmax over the two classes (`IsNext` and `NotNext`).
+
+3. The input embeddings are passed through multiple encoder layers to output some final hidden states. To use the pretrained model for classification, add a sequence classification head on top of the base BERT model. The sequence classification head is just a linear layer that accepts the final hidden states and does a linear transformation to convert them into logits. The cross-entropy loss is then calculated between the logits and target to find the most likely label.
```

Comment:
This is not text classification but masked language modeling or BERT pretraining. Like you did for Wav2Vec2, there should be a section after the pretraining one that details how to get to text classification from the pretrained model.